### ISSUE TITLE
swRho() not setting dim of result properly

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1619]
e.g.
```R
swRho(S,T,p)
```
with `S` and `T` being arrays and `p` being a single value ... it ought to give an array but it does not.

I'll look at this later today.

**Comment by dankelley** [comment]
Hm, I guess an array isn't a matrix, so the following did not catch this. I'll want to update this throughout `R/sw.R`.
```R
    Smatrix <- is.matrix(l$salinity)
```

**Comment by dankelley** [comment]
This is fixed in commit 7fc3706f528ea5093f4f6cd30a7a14aefc1f317b of the "develop" branch.  I added the following to the test suite, so we know that this works for both arrays and matrices, and for both the 'eos' and 'unesco' equations of state.

```R
test_that("swRho handles matrix and array data", {
          data(ctd)
          lon <- ctd[["longitude"]]
          lat <- ctd[["latitude"]]
          S <- ctd[["salinity"]][1:20]
          T <- ctd[["temperature"]][1:20]
          p <- ctd[["pressure"]][1:20]
          Sm <- matrix(S, nrow=10)
          Tm <- matrix(T, nrow=10)
          pm <- matrix(p, nrow=10)
          Sa <- array(S, dim=c(2, 2, 5))
          Ta <- array(T, dim=c(2, 2, 5))
          pa <- array(p, dim=c(2, 2, 5))
          ## 'unesco' equation of state
          rho <- swRho(S, T, p, eos="unesco")
          expect_equal(swRho(Sm, Tm, pm, eos="unesco"),
                       matrix(rho, nrow=10))
          expect_equal(swRho(Sa, Ta, pa, eos="unesco"),
                       array(rho, dim=c(2, 2, 5)))
          ## 'gsw' equation of state
          rho <- swRho(S, T, p, longitude=lon, latitude=lat, eos="gsw")
          expect_equal(swRho(Sm, Tm, pm, longitude=lon, latitude=lat, eos="gsw"),
                       matrix(rho, nrow=10))
          expect_equal(swRho(Sa, Ta, pa, longitude=lon, latitude=lat, eos="gsw"),
                       array(rho, dim=c(2, 2, 5)))
})
```
